### PR TITLE
GH-90699: fix refleak in `_testcapimodule.c`

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-90699

### CODE REVIEW

File: Modules/_testcapimodule.c [hunk 1]
```diff
@@ -3382,6 +3382,7 @@ slot_tp_del(PyObject *self)
         else
             Py_DECREF(res);
     }
+    Py_DECREF(tp_del);
```
Comment (corona10) [inline Modules/_testcapimodule.c]: nit How about move this code into just after `del = _PyType_Lookup(Py_TYPE(self), tp_del);`?

**Review by corona10 — COMMENTED** [review]
I check the leak and solution through
`````
./python.exe -m test test_finalization -R 3:3
Raised RLIMIT_NOFILE: 256 -> 1024
0:00:00 load avg: 3.73 Run tests sequentially
0:00:00 load avg: 3.73 [1/1] test_finalization
beginning 6 repetitions
123456
......

== Tests result: SUCCESS ==

1 test OK.

Total duration: 1.0 sec
Tests result: SUCCESS
``````

But please take a look at https://github.com/python/cpython/pull/99236#discussion_r1016178622 nit message.

File: Modules/_testcapimodule.c (hunk 1 above)
Comment (kumaraditya303) [reply Modules/_testcapimodule.c]: done

**Review by corona10 — APPROVED** [review]
lgtm!

**Pull request merged**